### PR TITLE
crtMthds ArrayList

### PR DESCRIPTION
**Pull request by ConfusedSatlan**

### CODE REVIEW

**Review by Dimagaa — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
code style check failed and your student colleague haven't approved PR

**Review by Dimagaa — CHANGES_REQUESTED** [review]

**Review by Dimagaa — APPROVED** [review]

File: src/main/java/core/basesyntax/ArrayList.java [hunk 1]
```diff
@@ -1,48 +1,113 @@
...
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); ++i) {
+            add(list.get(i));
+        }
+    }
 
+    private void increase() {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/ArrayList.java:38]: move all private methods below public

File: src/main/java/core/basesyntax/ArrayList.java [hunk 2]
```diff
@@ -1,48 +1,113 @@
...
-
+        checkIndex(index);
+        elementArray[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        final T rmElement = (T) elementArray[index];
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/ArrayList.java:60]: bad variable name

File: src/main/java/core/basesyntax/ArrayList.java [hunk 3]
```diff
@@ -1,48 +1,113 @@
...
+        }
+    }
+
+    private void checkIndexAdd(int index) {
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index " + index + " out of bounds ");
+        }
+    }
+
+    private int findObject(T object) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/ArrayList.java:100]:
```suggestion
    private int findIndex(T object) {
```

File: src/main/java/core/basesyntax/ArrayList.java [hunk 4]
```diff
@@ -1,48 +1,113 @@
...
+            if (equals(elementArray[i], object)) {
+                return i;
+            }
+        }
+        throw new NoSuchElementException("This " + object + "does not exist in array");
+    }
+
+    private boolean equals(T firstObject, T secondObject) {
+        return (firstObject == secondObject)
+                || (firstObject != null && firstObject.equals(secondObject));
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/ArrayList.java:111]:
```suggestion
        return firstObject == secondObject
                || firstObject != null && firstObject.equals(secondObject);
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/ArrayList.java (hunk 1 above)
Comment (nacenik) [reply src/main/java/core/basesyntax/ArrayList.java:70]: not fixed.

File: src/main/java/core/basesyntax/ArrayList.java [hunk 5]
```diff
@@ -1,48 +1,113 @@
...
+            }
+        }
+        throw new NoSuchElementException("This " + object + "does not exist in array");
+    }
+
+    private boolean equals(T firstObject, T secondObject) {
+        return firstObject == secondObject
+                || firstObject != null && firstObject.equals(secondObject);
+    }
+
```
Comment (nacenik) [inline src/main/java/core/basesyntax/ArrayList.java:46]: All private methods should be after all publics

File: src/main/java/core/basesyntax/ArrayList.java [hunk 6]
```diff
@@ -1,48 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE_ARRAY = 10;
+    private static final double INCREASE_SIZE_NUM = 1.5;
```
Comment (nacenik) [inline src/main/java/core/basesyntax/ArrayList.java:7]:
```suggestion
    private static final double INCREASE_SIZE_NUMBER = 1.5;
```

**Review by nacenik — CHANGES_REQUESTED** [review]
Good solution, let's improve it

**Review by nacenik — APPROVED** [review]
LGTM! 👍

**Pull request closed without merging**